src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,190 @@
+package nextstep.subway.line.domain;
+
+import nextstep.subway.station.domain.Station;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.EntityNotFoundException;
+import javax.persistence.OneToMany;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Embeddable
+public class Sections {
+
+    private static final String CAN_NOT_FIND_FIRST_SECTION = "can not find first section.";
+    private static final String BOTH_ALREADY_ADDED = "both stations is already added.";
+    private static final String BOTH_NOT_CONTAINS = "both stations is not contains.";
+    private static final String SECTIONS_HAS_ONLY_ONE = "sections has only one.";
+    private static final String NOT_CONTAINS_STATION = "not contains station.";
+    private static final int REMOVABLE_SIZE = 2;
+
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+
+    public Sections() {
+    }
+
+    public Sections(List<Section> sections) {
+        this.sections = sections;
+    }
+
+    public List<Section> getSections() {
+        return sections;
+    }
+
+    public void add(Section section) {
+        if (isFirstNewSection()) {
+            sections.add(section);
+            return;
+        }
+        addNotFirstNewSection(section);
+    }
+
+    private boolean isFirstNewSection() {
+        return sections.isEmpty();
+    }
+
+    private void addNotFirstNewSection(Section newSection) {
+        checkAddable(newSection);
+        sections.forEach(section -> section.update(newSection));
+        sections.add(newSection);
+    }
+
+    private void checkAddable(Section section) {
+        List<Station> stations = getSortedStation();
+        if (isBothAdded(section, stations)) {
+            throw new IllegalArgumentException(BOTH_ALREADY_ADDED);
+        }
+        if (isBothNotContains(section, stations)) {
+            throw new IllegalArgumentException(BOTH_NOT_CONTAINS);
+        }
+    }
+
+    private boolean isBothAdded(Section section, List<Station> stations) {
+        return stations.containsAll(section.getUpDownStation());
+    }
+
+    private boolean isBothNotContains(Section section, List<Station> stations) {
+        return !stations.contains(section.getUpStation()) && !stations.contains(section.getDownStation());
+    }
+
+    public List<Station> getSortedStation() {
+        List<Section> sortedSection = getSortedSection();
+        LinkedHashSet<Station> sortedSet = sortedSection.stream()
+                .flatMap(section -> section.getUpDownStation().stream())
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+        return new ArrayList<>(sortedSet);
```

Comment:
```suggestion
        return getSortedSection().stream()
            .flatMap(section -> section.getUpDownStation().stream())
            .distinct()
            .collect(Collectors.toList());
```
극단적으로 줄여본다면 이렇게도 가능하겠네요 :)